js/client/modules/@arangodb/inspector.js

```diff
@@ -594,33 +595,116 @@ function getServerData(arango) {
-          tmp = executeExternalAndWait(
-            '/bin/bash', ['-c', 'df -h | tee /tmp/inspector-df.out > /dev/null']);
-          const df = fs.readFileSync('/tmp/inspector-df.out', 'utf8').slice(0,-1);
-          tmp = executeExternalAndWait(
-            '/bin/bash', ['-c', 'cat /proc/meminfo | tee /tmp/inspector-meminfo.out > /dev/null']);
-          const meminfo = fs.readFileSync('/tmp/inspector-meminfo.out', 'utf8').slice(0,-1);
-          tmp = executeExternalAndWait(
-            '/bin/bash', ['-c', 'uptime | tee /tmp/inspector-uptime.out > /dev/null']);
-          const uptime = fs.readFileSync('/tmp/inspector-uptime.out', 'utf8').slice(0,-1);
+          var uname, date, dmesg, df, meminfo, cpuinfo, uptime, release, top, du,
+              docker = false; 
+
+          // Operating system calls
+
+          const prefix =
+                '/tmp/inspector-' +
+                Math.random().toString(36).substring(2, 15) +
+                Math.random().toString(36).substring(2, 15) + '-';
           tmp = executeExternalAndWait(
-            '/bin/bash', ['-c', 'uname -a | tee /tmp/inspector-uname.out > /dev/null']);
-          const uname = fs.readFileSync('/tmp/inspector-uname.out', 'utf8').slice(0,-1);
-          var top;
-          if (status.pid !== undefined) {
+            '/bin/sh', ['-c', 'uname -a | tee ' + prefix + 'uname.out > /dev/null']);
+
+          var unix = tmp.exit === 0;
+          var linux = false, osx = false;
+          
+          if (unix) { // For unix falvours we can do better
+
+            var cli = '/bin/sh';
+            // OS name etc
+            uname = fs.readFileSync(prefix + 'uname.out', 'utf8').slice(0,-1);
+            linux = uname.split(' ')[0] === 'Linux';
+            osx = uname.split(' ')[0] === 'Darwin';
+
+            var tmp = executeExternalAndWait(
+              cli, ['-c', 'cat /proc/1/cgroup | grep "0::" | grep docker']);
+            docker = tmp.exit == 0;
+              
+            // System time
             tmp = executeExternalAndWait(
-              '/bin/bash', ['-c', 'top -b -H -p ' + status.pid + ' -n 1 | tee /tmp/inspector-top.out > /dev/null']);
-            top = fs.readFileSync('/tmp/inspector-top.out', 'utf8').slice(0,-1);
-          }
+              cli, ['-c', 'date -u "+%Y-%m-%d %H:%M:%S %Z" | tee '
+                            + prefix + 'date.out > /dev/null']);
+            date = fs.readFileSync(prefix + 'date.out', 'utf8').slice(0,-1);
+
+            // Linux kernel messages
+            if (linux) {
+              tmp = executeExternalAndWait(
+                cli, ['-c', 'dmesg | tee ' + prefix + 'dmesg.out > /dev/null']);
+              dmesg = fs.readFileSync(prefix + 'dmesg.out', 'utf8').slice(0,-1);
+            }
 
+            // Disk fill status
+            tmp = executeExternalAndWait(
+              cli, ['-c', 'df -h | tee ' + prefix + 'df.out > /dev/null']);
+            df = fs.readFileSync(prefix + 'df.out', 'utf8').slice(0,-1);
```

Comment:
rather use `fs.join(...)` to construct filenames... (also applies for the following filenames)